astropy/wcs/tests/test_wcs.py

```diff
@@ -118,7 +118,7 @@ def test_fixes():
     with pytest.raises(wcs.InvalidTransformError), pytest.warns(wcs.FITSFixedWarning) as w:
         wcs.WCS(header, translate_units='dhs')
 
-    if _WCSLIB_VER >= Version('7.4'):
+    if _WCSLIB_VER == Version('7.4'):
```

Comment:
```suggestion
    if Version('7.4') <= _WCSLIB_VER < Version('7.6'):
```
else would fail for (system) WCSLIB 7.5 – see 1ec8c86 for the other corresponding instances.
Depending if we want to ensure that `ASTROPY_USE_SYSTEM_WCSLIB=1` builds are testing in the same way.

Reply:
I slightly disagree with this approach. The reason is that we have never bundled `7.5` with `astropy` not even for a short time on `master` / `main`. I cannot be sure that `7.5` was issuing the same exact number of warnings  as `7.4` with the same wording.

Unless... you have played with `7.5` and run all the tests that were passing on `7.4` and you can confirm that number of warnings and warning message is exactly the same in `7.5` as it is in `7.4`.

Maybe we should use `skipif` when WCSLIB `7.5` (which only can be used as stand-alone) is used for testing.

Reply:
> Unless... you have played with `7.5` and run all the tests that were passing on `7.4` and you can confirm that number of warnings and warning message is exactly the same in `7.5` as it is in `7.4`.

Yes, I have tested it with the incremental update to 7.5 in f333e84 as part of #11514, and that passed all those tests.
But of course we could also take the position that we don't officially support 7.5 and will not test it.
Seems reasonable to require `>= 7.6` for 4.3 anyway, but then there are also still all the test cases for versions back to 5.x or even earlier, and I don't think anyone _should_ use those.

Reply:
@dhomeier I cherry picked those 2 commits as well as added support for testing with external WCSLIB 7.5.